modules/system/assets/js/framework.js

```diff
@@ -415,7 +431,24 @@ if (window.jQuery.request !== undefined) {
         }
     }
 
-    $(document).on('change', 'select[data-request], input[type=radio][data-request], input[type=checkbox][data-request]', function documentOnChange() {
+    // Appends an object to the form data
```

Comment:
This serializes a deep object to form name and value pairs. If the index is numeric and the value is not an object then the name is appended with [], otherwise it is appended with [index] and `appendObjectToFormData` is called recursively on the value.